Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -311,11 +311,16 @@ bool AuxiliaryProcessProxy::dispatchSyncMessage(IPC::Connection& connection, IPC
     return m_messageReceiverMap.dispatchSyncMessage(connection, decoder, replyEncoder);
 }
 
-void AuxiliaryProcessProxy::didFinishLaunching(ProcessLauncher*, IPC::Connection::Identifier connectionIdentifier)
+void AuxiliaryProcessProxy::didFinishLaunching(ProcessLauncher* launcher, IPC::Connection::Identifier connectionIdentifier)
 {
     ASSERT(!m_connection);
     ASSERT(isMainRunLoop());
 
+#if USE(EXTENSIONKIT)
+    if (launcher)
+        launcher->releaseLaunchGrant();
```

Comment:
We're releasing the grant too early here. The ProcessThrottler hasn't had a chance to take a process assertion yet, thus, the child process will likely suspend and resume shortly after.

The ProcessThrottler only gets a chance to take a process assertion after `ProcessThrottler::didConnectToProcess()` has been called (because that's when it gets the PID). `didConnectToProcess()` get called later on, in subclasses' didFinishLaunching().

We should probably refactor this code a bit so that we release the grant *after* the process throttler has had a chance to take an assertion. BTW, I think AuxiliaryProcessProxy could in theory notify the ProcessThrottler itself nowadays. It used to be that the AuxiliaryProcessProxy didn't know about the ProcessThrottler. However, I believe we changed that a while back.

Reply:
Ah, good point! In order to make the changes minimal now, would you be ok with calling `releaseLaunchGrant` in each process proxy?

Thanks for reviewing!